src/utils.jl

```diff
@@ -100,6 +100,6 @@ function md_hlines(md)
 end
 
 function strlimit(str::AbstractString, limit = 30)
-  str = lastindex(str) > limit ?  str[1:prevind(str, limit)]*"…" : str
+  str = lastindex(str) > limit ? str[1:prevind(str, limit)]*" ..." : str
```

Comment:
` ...` would be more consistent with the same stuff from the usual Atom packages, like autocomplete-plus and so.

Reply:
Honestly, I prefer `…` here -- this should guarantee that your string is *never* longer than `limit`.

Reply:
ah yeah, I understand.

Apart from that, I think this function is better to respect the _length_ of `str` instead of code units of that ?

I mean the implementation below would behave more intuitively imho:
```jl
function strlimit(str::AbstractString, limit::Int = 30, ellipsis::AbstractString = " ...")
  charary = collect(str)
  str = length(charary) > limit ? join(charary[1:limit - length(ellipsis)])*ellipsis : str
  filter(isvalid, str)
end
```

then `strlimit("jμλια", 5) == "jμλια"` (the current implementation would return `jμλ…` instead).

Reply:
Ok, then let's go with 
```
function strlimit(str::AbstractString, limit::Int = 30, ellipsis::AbstractString = "…")
  io = IOBuffer()
  i = 0
  for c in str
    i > limit - length(ellipsis) && break
    isvalid(c) || continue
    
    print(io, c)
    i += 1
  end
  length(str) > limit && print(io, ellipsis)
  
  return String(take!(io))
end
```
instead -- that should be correct in all cases and also have better performance than the array manipulations.

Reply:
ah yeah, it should be faster, thanks for the suggestion :)

fixed with 51fb20d ;)

Reply:
The problem with the `enumerate` solution is that you might discard too many chars:
```
julia> length(strlimit2(String(rand(UInt8, 100)), 20)) # this is my implementation
20

julia> length(strlimit(String(rand(UInt8, 100)), 20))
13
```

My version still isn't entirely correct, so let's use
```
function strlimit2(str::AbstractString, limit::Int = 30, ellipsis::AbstractString = "…")
  will_append = length(str) > limit
  
  io = IOBuffer()
  i = 1
  for c in str
    will_append && i > limit - length(ellipsis) && break
    isvalid(c) || continue
    
    print(io, c)
    i += 1
  end
  will_append && print(io, ellipsis)
  
  return String(take!(io))
end
```
instead :) Would also be great if you could add a test or two, but that doesn't need to hold up this PR.

Reply:
Aaah, ashamed, thanks for the fix again, now I see why `i` is needed ...

***

> Would also be great if you could add a test or two, but that doesn't need to hold up this PR.
Yeah, would be great. I may try to add tests, even tho I'm not sure I can come up with good ones, but the one using `rand` seems cool to me.

Reply:
> Aaah, ashamed, thanks for the fix again, now I see why i is needed ...

No worries, I had your exact version at first too.

> Yeah, would be great. I may try to add tests, even tho I'm not sure I can come up with good ones, but the one using rand seems cool to me.

Maybe also something simple like 
```
@test strlimit("jμλια", 5) == "jμλια"
@test strlimit("jμλια", 4) == "jμλ…"
@test strlimit("abcde", 5) == "abcde"
@test strlimit("abcde", 4) == "abc…"
@test strlimit("jμ"*String(b"\xb6")*"iλια", 5) == "jμλια" # this probably doesn't pass though
```
etc

Reply:
Hope these are addressed within the latest commit ;)